Feature/create tables

ulises2607 opened the pull request:
- Created a schema based on the exercise diagram.

ulises2607 committed:
Add patients, medical_histories and invoices tables

Felipe-Perez-Ferraro committed:
Created invoice_items and treatments tables

thasipablo committed:
Update schema_based_on_diagram.sql

Bantealem commented on schema_based_on_diagram.sql:
```diff
@@ -0,0 +1,36 @@
+    type VARCHAR(60),
+    name VARCHAR(60)
+);
+
```
Great job so far. 
1. Kindly, create a join table for a `many-to-many` relationship.  Based on the diagram `medical_histories` and `treatments` table has a `many-to-many` relationship. It is also one of the project requirements.  When you create the table don't forget the PRIMARY KEY. Kindly, create it.
![image](https://github.com/ulises2607/Clinic-Database/assets/101523582/b382d387-3a4e-4728-a18d-981c5ed0f66f)

2. Please create indexes for all FOREIGN KEYS. It helps the data optimization and integrity. It is also one of the project requirements.  Kindly, create it.
![image](https://github.com/ulises2607/Clinic-Database/assets/101523582/fb204c45-855d-4bef-98b5-89e2d1cc88e4)

Bantealem reviewed (changes requested):
# Status: Change Required
Hi @ulises2607, @Felipe-Perez-Ferraro and @thasipablo  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] Correct Git flow
- [x] Professional Readme

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Felipe-Perez-Ferraro committed:
Added join table and the indexes for FK

shahadat3669 commented on schema_based_on_diagram.sql:
```diff
@@ -0,0 +1,49 @@
+    quantity INTEGER NOT NULL,
+    total_price DECIMAL(10,2) NOT NULL,
+    invoice_id INTEGER REFERENCES invoices(id),
+    treatment_id INTEGER REFERENCES treatments(id)
```
- [ ] [Optioanl] 👍 Good progress on your code! However, there's a small issue that I've noticed. When running your entire file, it throws an error because you're using the `id` of the `treatments` table before creating the `treatments` table itself. To resolve this issue, consider moving the creation of the `treatments` table before the line where you reference the `treatments` table's `id` in the foreign key constraint. This way, the table will exist when the foreign key is created. Keep up the good work! 🛠️👏

![image](https://github.com/ulises2607/Clinic-Database/assets/55840999/5599cc52-e373-4c2a-aa8c-8266a03b90ee)

shahadat3669 commented on schema_based_on_diagram.sql:
```diff
@@ -0,0 +1,49 @@
+
+-- Indexes
+CREATE INDEX idx_patient_id ON medical_histories(patient_id);
+CREATE INDEX idx_medical_history_id ON invoices(medical_histories_id);
```
- [ ] 👍 Great work on creating the index for the `medical_history_id` in the `invoices` table! However, there seems to be a small typo error in your code. You've used `medical_histories_id` instead of `medical_history_id`. To ensure accuracy, please consider updating your code to resolve this typo error. Keep up the good work! 👏🛠️

![image](https://github.com/ulises2607/Clinic-Database/assets/55840999/0ff6f263-225c-4396-b698-8b3b4a617425)

shahadat3669 reviewed (changes requested):
Hi everyone, 

Great job so far! 👏 You've done well, and you're almost there for the next project ⏭️, but there are a few issues 🐛 that you still need to work on.

<img  src="https://media2.giphy.com/media/m3864rBwwBTKMAbpn3/giphy.gif" />


## Don't Worry 🙏

It's okay to have bugs in your code. 🆗 Please see this code review as an opportunity to learn practical coding skills and discover new tips that can improve your code quality 💯 and help you **get even better** 💼.

**I will guide you on how to address these areas down below 👇.**

## Highlights
✅ new tables created in schema_based_on_diagram.sql file.
✅ all tables from the diagram are created (by using `CREATE TABLE` command)
✅ all "join table" for the many-to-many relastionships is created


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- x
- y
- z

Cheers and Happy coding!👏👏👏

## Have any doubt ❓

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

<table width="100%" align="center">
  <tr>
    <td colspan="4" align="center">
      <strong>Contact me: Shahadat Hossain</strong>
    </td>
  </tr>
  <tr>
    <td align="center">
      <a href="https://github.com/shahadata3669">
        <img src="https://img.shields.io/badge/github-%23121011.svg?style=for-the-badge&logo=github&logoColor=white" alt="GitHub">
      </a>
    </td>
    <td align="center">
      <a href="https://www.linkedin.com/in/shahadat-cseng/">
        <img src="https://img.shields.io/badge/linkedin-%230077B5.svg?style=for-the-badge&logo=linkedin&logoColor=white" alt="LinkedIn">
      </a>
    </td>
    <td align="center">
      <a href="https://app.slack.com/client/T47CT8XPG/D05GT4E4WAY/rimeto_profile/U04426SD71V">
        <img src="https://img.shields.io/badge/Slack-4A154B?style=for-the-badge&logo=slack&logoColor=white" alt="Slack">
      </a>
    </td>
    <td align="center">
      <a href="mailto:shahadat3669@gmail.com?subject=Microverse%20Code%20Review">
        <img src="https://img.shields.io/badge/Gmail-D14836?style=for-the-badge&logo=gmail&logoColor=white" alt="Email">
      </a>
    </td>
  </tr>
</table>

## ⚠️ WARNING ⚠️

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

thasipablo committed:
Update schema_based_on_diagram.sql

- Move treatment table before it is referenced. 
- Correct typo in `medical_history_id` index.

thasipablo commented:
Thank you @shahadat3669 
We fixed these issues.

t-yanick reviewed (approved):
# STATUS: MILESTONE APPROVED :1st_place_medal: 

Hi @ulises2607, @thasipablo and @Felipe-Perez-Ferraro  

### Great job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Requested changes implemented 👍 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  
Congratulations! 🎉 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

thasipablo commented:
Thank you @t-yanick
